File: src/Sentry/Platforms/Android/Java/Buggy.java
package io.sentry.android.supplemental;

public final class Buggy {
    private Buggy() {}

    public static void throwRuntimeException(String msg) {
        throw new RuntimeException(msg);
    }

    public static void throwRuntimeExceptionOnBackgroundThread(String msg) {
        Thread thread = new Thread() {
            public void run() {
                throw new RuntimeException(msg);
            }
        };

        thread.start();
    }
}

File: samples/Sentry.Samples.Android/Kotlin/buggy.kt
package io.sentry.samples.android.kotlin

import kotlin.concurrent.thread

object Buggy {
    @JvmStatic fun `throw`() {
        try {
            throw Exception("Bugs in Kotlin 🐛")
        }
        catch (e: Exception) {
            throw e
        }
    }
    @JvmStatic fun throwOnBackgroundThread() {
        thread(start = true) {
            throw Exception("Kotlin 🐛 from a background thread.")
        }
    }
}
